### ISSUE TITLE
Changing frozen status when finetuning

### ISSUE THREAD

**Issue by mikerossgithub** [allenai/allennlp#2562]
- OS: [e.g. OSX, Linux]
linux
 - Python version: [e.g. 3.6.1]
3.6.7
 - AllenNLP version: [e.g. v0.8.2, or "I installed from master"]
0.8.2
 - PyTorch version: (if you installed it yourself)
1.0.0

**Question**
I have trained a model using a frozen bert token embedder (`pretrained-bert`). Now I want to fine-tune from where I finished training, but with the embedder layers unfrozen. For normal training, this is specified with `requires_grad=True` in the `model...token_embedders` section of the config. But the `model` section is ignored during fine-tuning, and the original values  `requires_grad==False` seem to be persisted in the weights file.

My interim solution was to overload `_load` for my model and manually change all the bert layers to unfrozen. But I am wondering if there is a better way with the API. I see there is a "no_grad" regex, but not a "allow_grad" regex.

**Comment by schmmd** [comment]
Are you looking to override parameters in your model configuration?  You can accomplish this with `--overrides`.  Otherwise, could you re-ask your question?

**Comment by mayhewsw** [comment]
I've had a similar problem, and actually using overrides doesn't fix it. The fine-tune command will ignore any params from "model" (see [here](https://github.com/allenai/allennlp/blob/master/allennlp/commands/fine_tune.py#L199)) even if they come from overrides.

**Comment by mikerossgithub** [comment]
Thanks @schmmed

> Are you looking to override parameters in your model configuration? You can accomplish this with --overrides. Otherwise, could you re-ask your question?

Trainer allows you to use `no_grad` regex to freeze parameters which were otherwise unfrozen. 

https://github.com/allenai/allennlp/blob/53a46ab7227a7f4293030c06a1ad18d010e87d74/allennlp/training/trainer.py#L812

I'm just asking if there is a way to do the opposite, allow gradients for parameters which were otherwise frozen. I think adding an `allow_grad` regex would suffice? Should I submit a PR, or is there already a way to do this without altering AllenNlp code?

**Comment by matt-gardner** [comment]
The right way to do this going forward is to write a callback that runs at the start of training to freeze or unfreeze whatever parameters you want with custom code.  The callback trainer will be merged very soon.  Closing this in favor of that better way of doing things.
